fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/LongArithmethicFunctionKeyExpression.java

```diff
@@ -0,0 +1,283 @@
+         * to match the name of the function key expression.
+         *
+         * @param name the name of the function
+         * @param valueFunctionName the name of the associated {@link BuiltInFunction}
+         * @param operator a lambda representing function execution
+         * @return a new {@link Builder} of a binary arithmetic function expression
+         */
+        @Nonnull
+        public static Builder binaryFunction(@Nonnull String name, @Nonnull String valueFunctionName, @Nonnull BinaryOperator<Long> operator) {
+            return new Builder(name, valueFunctionName, 2, 2, null, operator);
+        }
+
+        /**
+         * Create a new builder of a function that can be either unary or binary.
+         *
+         * @param name the name of the function
+         * @param unaryOperator the function to execute if a single argument is provided
+         * @param binaryOperator the function to execute if two arguments are provided
+         * @return a new {@code Builder} of a function that can be unary or binary
+         */
+        @Nonnull
+        public static Builder bothFunction(@Nonnull String name, @Nonnull UnaryOperator<Long> unaryOperator, @Nonnull BinaryOperator<Long> binaryOperator) {
+            return new Builder(name, name, 1, 2, unaryOperator, binaryOperator);
+        }
+
+        /**
+         * Variant of {@link #bothFunction(String, UnaryOperator, BinaryOperator)} that supports using an alternate name
+         * for the {@link BuiltInFunction} used by the planner. By default, the value name will be assumed
+         * to match the name of the function key expression.
+         *
+         * @param name the name of the function
+         * @param valueFunctionName the name of the associated {@link BuiltInFunction}
+         * @param unaryOperator the function to execute if a single argument is provided
+         * @param binaryOperator the function to execute if two arguments are provided
+         * @return a new {@link Builder} of a function that can be unary or binary
+         */
+        @Nonnull
+        public static Builder bothFunction(@Nonnull String name, @Nonnull String valueFunctionName, @Nonnull UnaryOperator<Long> unaryOperator, @Nonnull BinaryOperator<Long> binaryOperator) {
+            return new Builder(name, valueFunctionName, 1, 2, unaryOperator, binaryOperator);
+        }
+    }
+
+    /**
+     * Factory for constructing built-in arithmetic functions that operate on {@code long}s.
+     */
+    @AutoService(FunctionKeyExpression.Factory.class)
+    public static class LongArithmethicFunctionKeyExpressionFactory implements FunctionKeyExpression.Factory {
+        @Nonnull
+        private static final List<FunctionKeyExpression.Builder> BUILDERS = ImmutableList.<FunctionKeyExpression.Builder>builder()
+                .add(Builder.binaryFunction(FunctionNames.ADD, (l, r) -> l + r))
```

Comment:
So no `mathExact()` here? I mean it's fine, just want to make sure it is what you intended.